src/rgw/driver/rados/rgw_putobj_processor.cc

```diff
@@ -341,8 +341,14 @@ int AtomicObjectProcessor::complete(size_t accounted_size,
                                     const char *if_nomatch,
                                     const std::string *user_data,
                                     rgw_zone_set *zones_trace,
-                                    bool *pcanceled, optional_yield y)
+                                    bool *pcanceled, 
+                                    const req_context& rctx)
 {
+  jspan span;
+  if (rctx.span) {
+    span = tracing::rgw::tracer.add_span("complete atomic operation", rctx.span->GetContext());
+  }
+
```

Comment:
these new spans aren't really part of the request context feature, can we move these to a separate pr?

it seems like we could emit a single span in `RGWRados::Object::Write::_do_write_meta()` to cover just the index transaction instead of emitting 3 different kinds of spans at this layer. the parent spans should know whether they're multipart or append operations

Reply:
> these new spans aren't really part of the request context feature, can we move these to a separate pr?
> 

ok. used it here mainly to verify that I'm passing the right value in the request context.
will move to a separate PR.

> it seems like we could emit a single span in `RGWRados::Object::Write::_do_write_meta()` to cover just the index transaction instead of emitting 3 different kinds of spans at this layer. the parent spans should know whether they're multipart or append operations

ok. will do that in the other PR

Reply:
@cbodley just to make sure that I understand the "complete()" API to which the `req_context` was added is invoked from `fetch_remote_obj()` not from `_do_write_meta()`.
I would add that to `_do_write_meta()` to write the span to the bi log, but i still need that in `fetch_remote_obj()` on the syncing side?

Reply:
> the "complete()" API to which the `req_context` was added is invoked from `fetch_remote_obj()` not from `_do_write_meta()`

each of these `*ObjectProcessor::complete()` functions are calling `obj_op.write_meta()` -> `_do_write_meta()`. you'd just need to pass in the `req_context` or span

Reply:
ok. got it